src/mame/juku/juku.cpp

```diff
@@ -413,19 +528,22 @@ void juku_state::pio0_portc_w(uint8_t data)
 
 	for (int i = 2; i < 8; i++)
 		if (BIT(data, i) !=  BIT(m_prev_portc, i))
-			logerror("pio0: c%d = %d\n", i, BIT(data, i));
+			osd_printf_verbose("pio0: c%d = %d\n", i, BIT(data, i));
 
 	floppy_image_device *floppy = m_floppy[BIT(data, 5)]->get_device();
 	m_fdc->set_floppy(floppy);
 
-	// the motor is always running for now
-	floppy->mon_w(0);
-
-//  m_floppy[0]->get_device()->ss_w(BIT(data, 6));
+	if (floppy)
+	{
+		// the motor is always running for now
+		floppy->mon_w(0);
+		floppy->ss_w(BIT(data, 6));
+	}
 
 	m_bank->set_bank(data & 0x03);
 
-	m_prev_portc = data;
+	if (m_prev_portc != data)
+		m_prev_portc = data;
```

Comment:
The test here doesn’t really provide any value – you want `m_prev_portc` to have the value of `data` at the end.